### ISSUE TITLE
neomutt 20231023 on Sonoma tries to convert some UTF-8 characters

### ISSUE THREAD

**Issue by tetsushiwahaha** [neomutt/neomutt#4084]
## Expected Behaviour

No conversion for UTF-8 multibyte characters expected.

## Actual Behaviour

Some Japanese multibyte 'comma,' 'dot,' and 'thee dot leader' are converted into ASCII characters.

Escpecially, 'three dot leader' is converted into '^[$ . .' Some mail reader cannot display it visible.

## Steps to Reproduce

After updating new macOS (Ventura -> Sonoma), 'brew upgrade' has been applied.

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded
  Which version did you use to use?

20220415 (probably) -> 20231023

My two Mac(M1, M2) machines have the same situation. 

## NeoMutt Version

```
NeoMutt 20231023
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 23.0.0 (arm64)
ncurses: ncurses 6.4.20221231 (compiled with 6.4.20221231)
libiconv: 1.11
libidn2: 2.3.4 (compiled with 2.3.4)
GPGME: 1.22.0
OpenSSL: OpenSSL 3.1.3 19 Sep 2023
libnotmuch: 5.6.0
PCRE2: 10.42 2022-12-11
storage: tokyocabinet, lmdb
compression: zlib

Configure options: --prefix=/opt/homebrew/Cellar/neomutt/20231023 --sysconfdir=/opt/homebrew/etc --autocrypt --gss --disable-idn --idn2 --lmdb --nls --notmuch --pcre2 --sasl --sqlite --tokyocabinet --zlib --with-lua=/opt/homebrew/opt/lua --with-ncurses=/opt/homebrew/opt/ncurses --with-ssl=/opt/homebrew/opt/openssl@3 --with-sqlite=/opt/homebrew/opt/sqlite

Compilation CFLAGS: -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/opt/homebrew/opt/libgpg-error/include -I/opt/homebrew/Cellar/libassuan/2.5.6/include -I/opt/homebrew/Cellar/gpgme/1.22.0/include -I/opt/homebrew/Cellar/neomutt/20231023/include -I/opt/homebrew/include/lua -I/opt/homebrew/Cellar/neomutt/20231023/include -I/opt/homebrew/Cellar/neomutt/20231023/include -I/opt/homebrew/opt/sqlite/include -I/opt/homebrew/Cellar/neomutt/20231023/include -DNCURSES_WIDECHAR -I/opt/homebrew/Cellar/neomutt/20231023/include -I/opt/homebrew/Cellar/openssl@3/3.1.3/include -I/opt/homebrew/Cellar/libidn2/2.3.4_1/include -I/opt/homebrew/Cellar/pcre2/10.42/include -I/opt/homebrew/opt/lmdb/include -I/opt/homebrew/Cellar/tokyo-cabinet/1.4.48/include -I/opt/homebrew/Cellar/neomutt/20231023/include -O2

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

コンパイル時オプション:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo -gnutls +gpgme 
  -gsasl +gss +hcache -homespool +idn -inotify -locales_hack +lua -mixmaster 
  +nls +notmuch +openssl +pcre2 +pgp +sasl +smime +sqlite +sun_attachment 
  +truecolor 

MAILPATH="/var/mail"
PKGDATADIR="/opt/homebrew/Cellar/neomutt/20231023/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/opt/homebrew/etc"
```

## Extra Info

* Operating System and its version: Sonoma 14.0
* Were you using multiple copies of NeoMutt at once? NO
* Were you using 'screen' or 'tmux'? NO
* Is your email local (maildir) or remote (IMAP)? REMOTE

**Comment by gahr** [comment]
Hi @tetsushiwahaha, where does the conversion happen? When you display a message you received in the pager? Or in the index?
Can you please provide a sample email?

**Comment by roccoblues** [comment]
Hi, I ran into the same issue during development. It's caused by a broken/incompatible version of libiconv in macOS Sonoma. I can be fixed by compiling Neomutt against libiconv shipped with Homebrew.

If you're comfortable compiling yourself you can use
```
-L/opt/homebrew/opt/ncurses/lib -L/opt/homebrew/Cellar/libiconv/1.17/lib
```

which should fix it.

Here's the relevant output from my build (notice the libiconv version difference):
````
./neomutt -v
NeoMutt 20231023
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 23.0.0 (arm64)
ncurses: ncurses 6.4.20221231 (compiled with 6.4.20221231)
libiconv: 1.17
libidn2: 2.3.4 (compiled with 2.3.4)
libnotmuch: 5.6.0
storage: lmdb
compression: zlib

Configure options: --disable-doc --disable-nls --testing --compile-commands --lmdb --zlib --debug --notmuch --with-notmuch=/opt/homebrew/Cellar/notmuch/0.38

Compilation CFLAGS: -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/opt/homebrew/Cellar/notmuch/0.38/include -I/usr/local/include -DNCURSES_WIDECHAR -I/usr/local/include -I/opt/homebrew/Cellar/libidn2/2.3.4_1/include -I/opt/homebrew/opt/lmdb/include -I/usr/local/include -MJ$(@:.o=.o.json) -O2
````

@gahr @flatcap do you know who's responsible for the Homebrew formular?

**Comment by tetsushiwahaha** [comment]
> Hi @tetsushiwahaha, where does the conversion happen? When you display a message you received in the pager? Or in the index? Can you please provide a sample email?

In the pager... hard coded. 
I am afraid that Japanese text can be visible in this interface...
Anyway let me try:

突破しまた^[$ ..大学自身がオワコンにならないよう,気を引き締めておかないといけないですね^[$ ..

**Comment by gahr** [comment]
> @gahr @flatcap do you know who's responsible for the Homebrew formular?

According to https://github.com/Homebrew/homebrew-core/pull/149916, @Porkepix did the last update.

**Comment by tetsushiwahaha** [comment]
> Hi, I ran into the same issue during development. It's caused by a broken/incompatible version of libiconv in macOS Sonoma. I can be fixed by compiling Neomutt against libiconv shipped with Homebrew.
> 
> If you're comfortable compiling yourself you can use
> 
> ```
> -L/opt/homebrew/opt/ncurses/lib -L/opt/homebrew/Cellar/libiconv/1.17/lib
> ```
> 
> which should fix it.
> 
> Here's the relevant output from my build (notice the libiconv version difference):
> 
> ```
> ./neomutt -v
> NeoMutt 20231023
> Copyright (C) 1996-2022 Michael R. Elkins and others.
> NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
> NeoMutt is free software, and you are welcome to redistribute it
> under certain conditions; type 'neomutt -vv' for details.
> 
> System: Darwin 23.0.0 (arm64)
> ncurses: ncurses 6.4.20221231 (compiled with 6.4.20221231)
> libiconv: 1.17
> libidn2: 2.3.4 (compiled with 2.3.4)
> libnotmuch: 5.6.0
> storage: lmdb
> compression: zlib
> 
> Configure options: --disable-doc --disable-nls --testing --compile-commands --lmdb --zlib --debug --notmuch --with-notmuch=/opt/homebrew/Cellar/notmuch/0.38
> 
> Compilation CFLAGS: -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/opt/homebrew/Cellar/notmuch/0.38/include -I/usr/local/include -DNCURSES_WIDECHAR -I/usr/local/include -I/opt/homebrew/Cellar/libidn2/2.3.4_1/include -I/opt/homebrew/opt/lmdb/include -I/usr/local/include -MJ$(@:.o=.o.json) -O2
> ```
> 
> @gahr @flatcap do you know who's responsible for the Homebrew formular?

Thank you for your prompt reply. Let me try it

**Comment by SMillerDev** [comment]
> do you know who's responsible for the Homebrew formula?

The public is

**Comment by tetsushiwahaha** [comment]
I did brew upgrade today, and neomutt has been updated probably:
`==> Upgrading neomutt
  20231023 -> 20231103 

==> Pouring neomutt--20231103.arm64_sonoma.bottle.tar.gz
🍺  /opt/homebrew/Cellar/neomutt/20231103: 89 files, 8.6MB
==> Running `brew cleanup neomutt`...
Removing: /opt/homebrew/Cellar/neomutt/20231023... (89 files, 8.6MB)
Removing: /Users/tetsushi/Library/Caches/Homebrew/neomutt--20231023... (2.7MB)
Removing: /Users/tetsushi/Library/Caches/Homebrew/neomutt--20231023.tar.gz... (3.6MB)
`
However, the problem which I reported in this thread seems not to be fixed. Still some UTF codes are converted into inappropriate ASCII characters. 

`% mutt -v
NeoMutt 20231103
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 23.1.0 (arm64)
ncurses: ncurses 6.4.20221231 (compiled with 6.4.20221231)
libiconv: 1.11
libidn2: 2.3.4 (compiled with 2.3.4)
GPGME: 1.23.1
OpenSSL: OpenSSL 3.1.4 24 Oct 2023
libnotmuch: 5.6.0
PCRE2: 10.42 2022-12-11
storage: tokyocabinet, lmdb
compression: zlib
`
In this message, libiconv is still 1.11, not 1.17. Although I did not try @roccoblues procedure yet, is this an only workaround?
